test/WebSites/TagHelpersWebSite/Views/Catalog_CacheTagHelper/Splash.cshtml

```diff
@@ -0,0 +1,8 @@
+﻿<h2>Category: @ViewBag.Category</h2>
+<h2>Region: @ViewBag.Locale</h2>
+<cache asp-vary-by-header="Locale" asp-vary-by-query="categoryid">
+    <h2>Cached content</h2>
+    @await Component.InvokeAsync("Splash")
+    @await Html.PartialAsync("_SplashPartial")
+    <div>CorrelationId in Splash: @ViewBag.CorrelationId</div>
```

Comment:
@yishaigalatzer \ @davidfowl, I'm not super sure if we decided on what happens if we call Flush inside of the tag. Currently that's not handled, but I suppose that's something we want to support? 

* Make it so that the content is buffered until the block is written and then invoke Flush. Keep track of this for the next time the tag is encountered and use that to flush for consecutive requests.
* Throw if you invoke Flush inside a `cache` tag.